server/data/prisonApiClient.ts

```diff
@@ -0,0 +1,38 @@
+import RestClient from './restClient'
+import { InmateDetail, VisitBalances } from './prisonApiTypes'
+import config from '../config'
+
+export const prisonApiClientBuilder = (token: string): PrisonApiClient => {
+  const restClient = new RestClient('prisonApi', config.apis.prison, token)
+  const prisonClient = new PrisonApiClient(restClient)
+
+  return prisonClient
+}
+
+class PrisonApiClient {
+  constructor(private readonly restclient: RestClient) {}
+
+  getOffender(offenderNo: string): Promise<InmateDetail> {
+    return this.restclient.get({
+      path: `/api/offenders/${offenderNo}`,
+    })
+  }
+
+  async getVisitBalances(offenderNo: string): Promise<VisitBalances | null> {
+    let balances: VisitBalances
+    try {
+      balances = await this.restclient.get({
```

Comment:
Could always `return` here, then change the last `return` on line 34 to just `return null`

Reply:
Then no need for `balances`

Reply:
Good suggestion, thanks 👍!